docs/dg/learningBasics.md

```diff
@@ -0,0 +1,140 @@
+1. Step into each break point when you think it is necessary to know more about the implementation, and step over otherwise to continue the execution.
+
+<box type="info" seamless>
+
+When tracing through the program execution, you can cross reference the architecture diagram and `javadoc` of the class and method to check your understanding of the procedure.
+</box>
+
+{{ step(3) }} Get started with contributing
+
+Here is a small [task](https://github.com/reposense/RepoSense/issues/1118) for you to get started on contributing to RepoSense back-end.
+
+* The basic idea is to change or remove the validation for the author name tag located in `AnnotatorAnalyzer.java` by modifying the`REGEX_AUTHOR_NAME_FORMAT` argument.
+* You may need to check that the unit and system testing still passes after making the change by running the command `gradlew systemtest` and `gradlew test`.
+
+Here is more explanation about the usage of author tag.
+
+{{ embed("Appendix: **Using `@@author` tags**", "../ug/usingAuthorTags.md") }}
+
+{{ step(4) }} Get familiar with the workflow
+
+1. The next step is to verify that your changes passes the testing and then submit the pull request. You can refer to [the _Workflow_ page](workflow.html).
+
+<!-- ==================================================================================================== -->
+
+## front-end
+
+<box type="info" seamless>
+
+This section is for developers who want to contribute to the back-end of RepoSense. You may skip this section if you want to contribute as a pure front-end developer.
+</box>
+
+The front-end implementation of RepoSense is located in `frontend/src`.
+
+{{ step(1) }} Learn the necessary tools
+
+It is necessary for you to learn the basics of `Vue.js`, `pug`, and `scss` before working on the project. 
+
+  {{ embed('Appendix: tools to learn for front-end development', 'tools.md', level=2) }}
+
+{{ step(3) }} Learn the RepoSense front-end architecture
+
+* You may want to refer to the [front-end architecture](report.html#report-architecture) to understand the relationship
+* Another way for you to understand the front is to use **Vue.js devtools** to inspect the how the various Vue Components interact with each other. You can refer to the [front-end debugging guide](workflow.html#debugging-front-end) for more information.
+
+{{ step(4) }} Get started with contributing
+
+Here is a small [task](https://github.com/reposense/RepoSense/issues/1268) for you to get started on contributing to RepoSense front-end.
+
+* The general goal is to eliminate the passing of variables that can be shared across different components, and move these variables to the **Vuex store**.
+* You can try to move `minDate`, `maxDate`, and `filterSearch` to **Vuex store**. 
+* When in doubt, you can refer to the pull request [#1319](https://github.com/reposense/RepoSense/pull/1319) and [#1320](https://github.com/reposense/RepoSense/pull/1320) to view the implementation detail.
```

Comment:
Let's move this into the issue itself with instructions on how and what to change.